### PR TITLE
SYCL kernel support for Pooling

### PR DESCRIPTION
**Pull request by lakkuushaphani**
# Description
Introducing SYCL backend for the oneDNN Pooling primitive. This contains support for Pooling functionality in both forward and backward propagations.

# Supported scope
Supported Data Types:
- Forward : f32, s32, f16, bf16, s8, u8
- Backward: f32, f16, bf16

## General

- [x] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [x] Have you formatted the code using clang-format?

## Tested Hardware
Nvidia Tesla T4

## Testing Scope
`benchdnn : passed`

Please refer the test output files
[Pooling_benchdnn_output.zip](https://github.com/oneapi-src/oneDNN/files/10948726/Pooling_benchdnn_output.zip)

### CODE REVIEW

**Review by densamoilov — CHANGES_REQUESTED** [review]
Please rebase it on the latest master.

File: src/gpu/nvidia/sycl_cuda_engine.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@
 #include "gpu/nvidia/sycl_cuda_stream.hpp"
 
 #include "gpu/sycl/ref_binary.hpp"
```
Comment (densamoilov) [inline src/gpu/nvidia/sycl_cuda_engine.cpp:41]: (random spot) Please add documentation.

**Comment by lakkuushaphani** [comment]
Reopening the PR due to accidental discard of the branch

File: src/gpu/nvidia/sycl_cuda_engine.cpp (hunk 1 above)
Comment (saisrujana) [reply src/gpu/nvidia/sycl_cuda_engine.cpp:42]: Yes, updated the Readme

**Comment by saisrujana** [comment]
> Please rebase it on the latest master.

Done.

**Review by densamoilov — APPROVED** [review]

**Pull request merged**